src/app/data-client.ts

```diff
@@ -38,6 +38,22 @@ interface TabularData {
   timeReceived?: Date;
 }
 
+interface TabularDataPoint {
+  partId: string;
+  resourceName: string;
+  resourceSubtype: string;
+  methodName: string;
+  timeCaptured: Date;
```

Comment:
per failing tests, I think this needs to be `Date | undefined` since it's possible no `timeCaptured` exists on the response.

Reply:
I didn't realize we had to account for that with protobuf! Noted